src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -60,7 +60,7 @@ private function __construct()
      *
      * @throws \InvalidArgumentException if $unprocessedHtml is anything other than a non-empty string
      */
-    public static function fromHtml(string $unprocessedHtml): AbstractHtmlProcessor
+    public static function fromHtml(string $unprocessedHtml): self
```

Comment:
Um, also for `::fromDomDocument` please :)

Reply:
Oh. Right. (Note to myself: Have coffee and breakfast first. Code after that, not before.)

Reply:
> Have coffee and breakfast first

Or as the newspaper-reading types do: look at the main article AND the side article :)